components/public-api-server/pkg/oidc/service.go

```diff
@@ -318,3 +323,25 @@ func (s *Service) CreateSession(ctx context.Context, flowResult *AuthFlowResult,
 	log.WithField("create-session-error", message).Error("Failed to create session (via server)")
 	return nil, message, fmt.Errorf("unexpected status code: %v", res.StatusCode)
 }
+
+func (s *Service) validateRequiredClaims(token *goidc.IDToken) error {
+	if len(token.Audience) < 1 {
+		return fmt.Errorf("audience claim is missing")
+	}
+	var claims struct {
+		Email string `json:"email,omitempty"`
+		Name  string `json:"name,omitempty"`
+		jwt.RegisteredClaims
+	}
+	err := token.Claims(&claims)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal claims of ID token: %w", err)
```

Comment:
[note to self] We should figure out how to encode HTTP errors in these handlers and return those directly from these methods to have an easier time returning the right error codes to the client.

Reply:
When propagated to frontend eventually, we wouldn't need the codes. From UX POV it would be get the unwrapped error, e.g. "email claim is missing". It's hard to render the long messages nicely.

Reply:
Not necessarily for the frontend, but we want metrics on the endpoints to know when something goes wrong. And for that, we need appropriate status codes else we can't distinguish user errors from system errors